tdms/cmake/version_from_git.cmake

```diff
@@ -0,0 +1,30 @@
+find_package(Git)
+if(GIT_FOUND)
+
+	# Get current branch name.
```

Comment:
What happens when one uses a snapshot, an archive of the source code without the git repository, e.g. https://github.com/UCL/TDMS/archive/refs/heads/201-versioning-in-cmake.tar.gz?

Reply:
Ha. Good spot. I thought it would call it `"development"` from the `#ifdef` in the C++. But it calls it `_` (underscore).

Perhaps more important, is how to fix the git repo-less tarballs one gets from the [releases page](https://github.com/UCL/TDMS/releases). Presumably, we really want **those** to be correctly versioned.

Reply:
I don't think that one can possibly work unless the version information is stored in a file checked in the repository, or unless you automatically create with cmake a `VERSION` file which is not checked in the repository but would be included a in "release tarball" that you upload in the release page on GitHub, but that wouldn't be the automatic source tarball generated by GitHub for the tag.

The alternative to use automatic git tagging is to write down the version number in the CMake file, and maybe use to add the git hash only when you're building straight out of a git repository.

Reply:
>  It obviously fails if the user untars to some other directory else.

Yeah, I don't like relying on the name of the parent directory, strictly speaking that's "outside" of the package (just slightly, it's on its edge).

> But I don't like hard-coding.

Well, if you want to make this work without a git repository this ***has*** to be hardcoded somewhere :slightly_smiling_face:

Reply:
Just to close the loop here (ancient post).
We'll hard-code the value as a fall back for now.